src/utils/Client.util.ts

```diff
@@ -274,3 +297,68 @@ export function getClientOptions(
     maxBodySizeMb: clientoptions?.maxBodySizeMb,
   };
 }
+
+/**
+ * Scheme, fqdn and port
+ */
+type EndpointTuple = [string, string, number];
+
+/**
+ * Parses an endpoint to scheme, fqdn and port
+ * @param address Endpoint address
+ * @returns EndpointTuple (scheme, fqdn, port)
+ */
+export function parseEndpoint(address: string): EndpointTuple {
+  let scheme = "http";
+  let fqdn = "localhost";
+  let port = 80;
+  let addr = address;
+
+  const addrList = address.split("://");
+
+  if (addrList.length === 2) {
+    // A scheme was explicitly specified
+    scheme = addrList[0];
+    if (scheme === "https") {
+      port = 443;
+    }
+    addr = addrList[1];
+  }
+
+  const addrParts = addr.split(":");
+  if (addrParts.length === 2) {
+    // A port was explicitly specified
+    if (addrParts[0].length > 0) {
+      fqdn = addrParts[0];
+    }
+    // Account for Endpoints of the type http://localhost:3500/v1.0/invoke
+    const portParts = addrParts[1].split("/");
+    port = parseInt(portParts[0], 10);
+  } else if (addrParts.length === 1) {
+    // No port was specified
+    // Account for Endpoints of the type :3500/v1.0/invoke
+    const fqdnParts = addrParts[0].split("/");
+    fqdn = fqdnParts[0];
+  } else {
+    // IPv6 address
+    const ipv6Parts = addr.split("]:");
```

Comment:
As unbelievable as this sounds, there is no official regex expression that can validate IPv6 addresses. I tried many, but none of them matched all possible valid addresses, so I decided to rely on the underlying client library to return an error on connection if the address is invalid. I didn't think this would justify adding a new dependency to the sdk.

Reply:
Makes sense, alrighty!